doc/getting_started/build_sw.md

```diff
@@ -1,217 +1,335 @@
+For example, building and testing the mask ROM UART driver unit tests:
 ```console
-bazel build //<path to build file>:<name of target>
+bazel test //sw/device/silicon_creator/lib/drivers:uart_unittest
 ```
 
-### Structure of our Bazel workspace
+## OpenTitan Bazel Workspace
 
-The rules for Bazel are described in a language called starlark, which looks a lot like Python.
+The rules for Bazel are described in a language called Starlark, which looks a lot like Python.
 
-The OpenTitan directory is defined as a Bazel workspace by the `//WORKSPACE` file.
+The `$REPO_TOP` directory is defined as a Bazel workspace by the `//WORKSPACE` file.
 `BUILD` files provide the information Bazel needs to build the targets in a directory.
 `BUILD` files also manage any subdirectories that don't have their own `BUILD` files.
 
-OpenTitan uses .bzl files to specify rules to build artifacts that require specific attention like target specific test rules and project specific binaries.
+OpenTitan uses .bzl files to specify custom rules to build artifacts that require specific attention like on-device test rules and project specific binaries.
 
-#### WORKSPACE file
+### WORKSPACE file
 
 The `WORKSPACE` file controls external dependencies such that builds can be made reproducible and hermetic.
-Bazel loads specific commits of external git repositories.
+Bazel loads specific external dependencies, such as various language toolchains.
 It uses them to build OpenTitan targets (like it does with bazel\_embedded) or to satisfy dependencies (as it does with abseil).
-To produce increasingly stable releases the `WORKSPACE` file will fix a number of these references to specific commits and not the default branch head.
-Dependencies will be added to the workspace so builds and tests are less sensitive to external updates and have a simpler getting started flow.
+To produce increasingly stable releases the external dependencies loaded in `WORKSPACE` file attempts to fix a all external `http_archive`s to a specific SHA.
+As we add more dependencies to the workspace, builds and tests will become less sensitive to external updates, and we will vastly simplify the [Getting Started]({{< relref "getting_started" >}}) instructions.
+
+### BUILD files
+
+Throughout the OpenTitan repository, `BUILD` files describe targets and dependencies in the same directory (and subdirectories that lack their own `BUILD` files).
+`BUILD` files are mostly hand-written.
+To maintain the invariant that hand-written files not be included in autogen directories, there are `BUILD` files that describe how to build and depend on auto-generated files in autogen subdirectories.
+
+## Linting Software
 
-#### BUILD files
+There are several Bazel rules that enable running quality checks and fixers on code.
+The subsections below describe how to use them.
+All of the tools described below are run in CI on every pull request, so it is best to run them before committing code.
```

Comment:
There are also two separate Python lint checks (mypy and flake8):
https://github.com/lowRISC/opentitan/blob/058344a4fc2a27460060718315407f4675fd5f00/ci/jobs/quick-lint.sh#L44-L50

AFAIK there aren't any Bazel rules to run these, but it seems worth mentioning here -- and I'll put in my vote for adding those rules at some point!

Reply:
Ah good point, there's actually a lot more quality rules missing, I filed an issue to track this: #13162. 

Side note, AFAIK, flake8 is not enforced in CI is it?